core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
I thought one of the main purpose of the shutdown hook is for resource disposal and cleanup? Why it is considered an anti-pattern?